lib/src/digest.dart

```diff
@@ -17,8 +17,22 @@ class Digest {
   /// This should be used instead of manual comparisons to avoid leaking
   /// information via timing.
   @override
-  bool operator ==(Object other) =>
-      other is Digest && const ListEquality().equals(bytes, other.bytes);
+  bool operator ==(Object other) {
+    if (other is Digest) {
+      List<int> a = bytes;
+      List<int> b = other.bytes;
+      if (a.length != b.length) {
+        return false;
```

Comment:
Document this behavior. An alternative reasonable thing is to keep running through both of them no matter how long they are although that is a little more likely to leak.

Reply:
Yeah, I considered it overkill, as I'll have to use the length of either `a` or `b`.